From 67aaf306c02b43120186625510557b449ce4cdba Mon Sep 17 00:00:00 2001
From: Carlos Leija <cileija@ti.com>
Date: Fri, 11 Mar 2011 02:12:46 -0600
Subject: OMAP4: Add call to omap4_secure_dispatcher early in CPU0 wake up

This will restore the kernel return address for HAL calls.

Change-Id: I77bfc639e95b7503f87cab9fb75ce46c57e93f8a
Signed-off-by: Carlos Leija <cileija@ti.com>
---
 arch/arm/mach-omap2/omap4-mpuss-lowpower.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-omap2/omap4-mpuss-lowpower.c b/arch/arm/mach-omap2/omap4-mpuss-lowpower.c
index 8a6fb06..2ac682f 100644
--- a/arch/arm/mach-omap2/omap4-mpuss-lowpower.c
+++ b/arch/arm/mach-omap2/omap4-mpuss-lowpower.c
@@ -773,6 +773,7 @@ cpu_prepare:
 	} else {
 		pwrdm_set_next_pwrst(cpu0_pwrdm, PWRDM_POWER_ON);
 		pwrdm_set_next_pwrst(mpuss_pd, PWRDM_POWER_ON);
+		omap4_secure_dispatcher(0x21, 4, 0, 0, 0, 0, 0);
 	}
 
 	/*
-- 
1.6.0.4

